File: c++/9342.cpp
#include <iostream>
#include <regex>
#include <string>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);
  int N;
  cin >> N;

  regex re("^[A-F]?A+F+C+[A-F]?$");

  for (int i = 0; i < N; i++) {
    string s;
    cin >> s;

    if (regex_match(s, re))
      cout << "Infected!";
    else
      cout << "Good";

    cout << '\n';
  }

  return 0;
}

File: js/9342.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = inputs[0];

const regex = /^[A-F]?A+F+C+[A-F]?$/;

for (let i = 1; i <= N; i++) {
  const s = inputs[i];

  if (regex.test(s)) console.log('Infected!');
  else console.log('Good');
}
